### ISSUE TITLE
Accessing attributes with a colon in the name

### ISSUE THREAD

**Issue by janaz** [mmckegg/json-query#20]
Attributes containing `:` character in their names are quite common in [HAL](http://stateless.co/hal_specification.html) JSON objects.

I noticed a regression in v2.0.1 which results in the attributes containing `:` not being correctly parsed. 

The following example illustrates the issue:
```js
var JsonQuery = require("json-query");

var obj = {
  "x:y": 123,
  "z": {
    "a:b": 456
  }
};

var x = JsonQuery("[x:y]", {data: obj}).value;
var y = JsonQuery("[z][a:b]", {data: obj}).value;
console.log(x, y);
```

**json-query v1.8.0**
> `123 456`

**json-query v2.0.1**

> `null null`

Is there any other way to access these attributes in v2.0.1?

**Comment by mmckegg** [comment]
Thanks for reporting. I was a little worried about this, hence the major release bump.

Will probably have to add some form of escaping. Either inside quotes, or 'slash' style.

Though for now, you can use this hack:
(there was a bug with this method in v2.0.1 but it is fixed in **v2.0.2**)

```js
var JsonQuery = require("json-query");

var obj = {
  "x:y": 123,
  "z": {
    "a:b": 456
  }
};

var helpers = {
  get: function (input, key) {
    return input[key]
  }
};

var x = JsonQuery(":get(x:y)", {data: obj, locals: helpers}).value;
var y = JsonQuery("[z]:get(x:y)", {data: obj, locals: helpers}).value;
console.log(x, y);
```

**Comment by mmckegg** [comment]
Another workaround (depending on your use case):

```js
JsonQuery(["[z][?]", "x:y"], {data: obj, locals: helpers}).value
```
